churaverse-plugins-client/src/gamePlugin/domain/baseGamePlugin.ts

```diff
@@ -0,0 +1,49 @@
+import { BasePlugin, IMainScene, Store, IEventBus } from 'churaverse-engine-client'
+import { GameIds } from '../interface/gameIds'
+
+export abstract class BaseGamePlugin extends BasePlugin<IMainScene> {
+  /** ゲーム一意のid */
+  protected readonly id: GameIds
+  /** ゲーム名 */
+  private readonly _name: string
```

Comment:
最初に代入された時点から変更がないのであれば、getterを作らずにそのままreadonlyとして良い気がします
`_hoge`のように置いてgetterを作るのは、外部から見たときにreadonly（定数）にしつつ、そのクラス内部では変更があるもの（変数）として扱うためです。
内部で変更する必要がないのであれば最初から定数のものを直接publicにするだけで良いかと思います
```suggestion
  public readonly name: string
```

Reply:
nameやparticipateIds, isMidwayParticipateなどは、値の変更はせず読み取りだけ行うため、`public readonly name: string`のように変更します。また、`participate`よりも`participant`の方が名詞として意味が理解しやすいと考えたため、修正します。